### PR TITLE
Update `upgraded_packages` get to be case & dash/hyphen insensitive

### LINKED ISSUE #63: pip packages should be case insensitive and `_` vs `-` are the same
There is a minor bug/corner case where the package name's case setting affects if it gets picked up.

From https://peps.python.org/pep-0426/#name

> All comparisons of distribution names MUST be case insensitive, and MUST consider hyphens and underscores to be equivalent.

Example:
- `Numpy` == `numpy` == `NuMPy`
- `My-package` == `my_package`

@ak-gupta or @NikhilJArora would you have time/interest to work on a fix for this?

### PR DESCRIPTION
**Pull request by NikhilJArora**
# edgetest version 2023.6.1

## Description

When generating the report, there is a method that gets the list of `upgraded_packages` for a given test environment.
The comparison of your `upgrade` list to your actual test environment (using `pip list`) is currently case sensitive even though `pip list` will always normalize your deps to lowercase with dashes.

This PR changes this comparison to be case and dash/hyphen insensitive to avoid missing upgraded deps due to case/dash differences between your upgrade list and actual env. See issue for more details.

Fixes #63

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing integration tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -481,3 +482,12 @@ def test_upgrade_pyproject_toml(tmpdir):
             "optional-dependencies": {"tests": ["pytest<=4.0.0,>=1.0.0"]},
         }
     }
+
+
+def test_isin_case_dashhyphen_ins():
+    vals = ["pandas", "python-dateutil"]
+
+    assert _isin_case_dashhyphen_ins("Pandas", vals)
```
Comment (fdosani) [inline tests/test_utils.py:491]: Looks great. Just one minor ask. Can we add some normal cases where its like `python-dateutil` to make sure `-` also come back the same

**Review by fdosani — COMMENTED** [review]
LGTM. just one small minor ask on the tests. 
TLDR: Just add a couple more tests where it is the same as expected to ensure `_isin_case_dashhyphen_ins` isn't messing up things which are fine.

File: tests/test_utils.py (hunk 1 above)
Comment (NikhilJArora) [reply tests/test_utils.py:491]: Updated!

**Review by fdosani — APPROVED** [review]
